<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>860</width>
    <height>704</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>lsystem drawer</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout_4">
    <property name="leftMargin">
     <number>10</number>
    </property>
    <property name="topMargin">
     <number>10</number>
    </property>
    <property name="rightMargin">
     <number>10</number>
    </property>
    <property name="bottomMargin">
     <number>10</number>
    </property>
    <item>
     <widget class="QGroupBox" name="groupBox">
      <property name="title">
       <string>Параметры</string>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <item>
        <layout class="QGridLayout" name="gridLayout_3">
         <item row="1" column="0">
          <widget class="QLabel" name="label_3">
           <property name="text">
            <string>Угол поворота</string>
           </property>
          </widget>
         </item>
         <item row="4" column="0">
          <widget class="QLabel" name="label_9">
           <property name="text">
            <string>Начальные координаты (x, y)</string>
           </property>
          </widget>
         </item>
         <item row="3" column="1">
          <layout class="QHBoxLayout" name="horizontalLayout_6">
           <item>
            <widget class="QSpinBox" name="spinBox_size_x">
             <property name="minimum">
              <number>1</number>
             </property>
             <property name="maximum">
              <number>1000</number>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSpinBox" name="spinBox_size_y">
             <property name="minimum">
              <number>1</number>
             </property>
             <property name="maximum">
              <number>1000</number>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item row="5" column="1">
          <widget class="QSpinBox" name="spinBox_start_angle">
           <property name="maximum">
            <number>360</number>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QSpinBox" name="spinBox_angle">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="maximum">
            <number>360</number>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="label_4">
           <property name="text">
            <string>На сколько углов делится плоскость</string>
           </property>
          </widget>
         </item>
         <item row="4" column="1">
          <layout class="QHBoxLayout" name="horizontalLayout_7">
           <item>
            <widget class="QSpinBox" name="spinBox_start_x">
             <property name="minimum">
              <number>-500</number>
             </property>
             <property name="maximum">
              <number>500</number>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSpinBox" name="spinBox_start_y">
             <property name="minimum">
              <number>-500</number>
             </property>
             <property name="maximum">
              <number>500</number>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item row="0" column="0">
          <widget class="QLabel" name="label_2">
           <property name="text">
            <string>Название системы:</string>
           </property>
          </widget>
         </item>
         <item row="3" column="0">
          <widget class="QLabel" name="label_8">
           <property name="text">
            <string>Размеры изображения (x, y)</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QLineEdit" name="lineEdit_name">
           <property name="placeholderText">
            <string>Стандатное название - lsystem</string>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QSpinBox" name="spinBox_plane_div">
           <property name="minimum">
            <number>2</number>
           </property>
           <property name="maximum">
            <number>360</number>
           </property>
          </widget>
         </item>
         <item row="5" column="0">
          <widget class="QLabel" name="label">
           <property name="text">
            <string>Начальный угол поворота</string>
           </property>
          </widget>
         </item>
         <item row="6" column="0">
          <widget class="QLabel" name="label_10">
           <property name="text">
            <string>Длина одного шага</string>
           </property>
          </widget>
         </item>
         <item row="6" column="1">
          <widget class="QSpinBox" name="spinBox_step_length">
           <property name="minimum">
            <number>1</number>
           </property>
           <property name="maximum">
            <number>500</number>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QGroupBox" name="groupBox_3">
         <property name="title">
          <string>Режим вычисления угла поворота</string>
         </property>
         <layout class="QHBoxLayout" name="horizontalLayout">
          <item>
           <widget class="QRadioButton" name="radioButton_angle_mode">
            <property name="text">
             <string>Угол</string>
            </property>
            <attribute name="buttonGroup">
             <string notr="true">buttonGroup_rotation_mode</string>
            </attribute>
           </widget>
          </item>
          <item>
           <widget class="QRadioButton" name="radioButton_div_mode">
            <property name="text">
             <string>Деление плоскости</string>
            </property>
            <property name="checked">
             <bool>true</bool>
            </property>
            <attribute name="buttonGroup">
             <string notr="true">buttonGroup_rotation_mode</string>
            </attribute>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_2">
         <item>
          <widget class="QLabel" name="label_6">
           <property name="text">
            <string>Аксиома (инициатор)</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="lineEdit_initiator"/>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QGroupBox" name="groupBox_5">
         <property name="title">
          <string>Теоремы</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_2">
          <item row="0" column="0">
           <widget class="QPlainTextEdit" name="plainTextEdit_rules">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
              <horstretch>1</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="placeholderText">
             <string>Теоремы системы по одной на строке</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <layout class="QVBoxLayout" name="verticalLayout">
         <item>
          <widget class="QLabel" name="label_5">
           <property name="text">
            <string>Шаг эволюции</string>
           </property>
          </widget>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_3">
           <item>
            <widget class="QSlider" name="horizontalSlider_step">
             <property name="minimum">
              <number>1</number>
             </property>
             <property name="maximum">
              <number>50</number>
             </property>
             <property name="pageStep">
              <number>1</number>
             </property>
             <property name="sliderPosition">
              <number>1</number>
             </property>
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="invertedAppearance">
              <bool>false</bool>
             </property>
             <property name="invertedControls">
              <bool>false</bool>
             </property>
             <property name="tickPosition">
              <enum>QSlider::TicksBelow</enum>
             </property>
             <property name="tickInterval">
              <number>100</number>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSpinBox" name="spinBox_step">
             <property name="minimum">
              <number>1</number>
             </property>
             <property name="maximum">
              <number>50</number>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_8">
         <item>
          <widget class="QPushButton" name="pushButton_ch_bg_color">
           <property name="text">
            <string>Выбрать цвет фона</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="pushButton_ch_line_color">
           <property name="text">
            <string>Выбрать цвет линий</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QPushButton" name="pushButton_update">
         <property name="text">
          <string>Обновить</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="pushButton_clear">
         <property name="text">
          <string>Очистить</string>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_5">
         <item>
          <widget class="QPushButton" name="pushButton_save_image">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
             <horstretch>1</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="text">
            <string>Сохранить результат</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="label_7">
           <property name="text">
            <string>Множитель масштаба</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="spinBox_scale">
           <property name="minimum">
            <number>1</number>
           </property>
           <property name="maximum">
            <number>1000</number>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QGroupBox" name="groupBox_2">
      <property name="title">
       <string>Результат</string>
      </property>
      <layout class="QGridLayout" name="gridLayout">
       <item row="0" column="1">
        <widget class="QLabel" name="label_result">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
           <horstretch>2</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="text">
          <string>TextLabel</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>860</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu">
    <property name="title">
     <string>Файл</string>
    </property>
    <addaction name="action_new"/>
    <addaction name="separator"/>
    <addaction name="action_open"/>
    <addaction name="separator"/>
    <addaction name="action_save"/>
    <addaction name="action_saveas"/>
    <addaction name="separator"/>
    <addaction name="action_exit"/>
   </widget>
   <widget class="QMenu" name="menu_2">
    <property name="title">
     <string>База данных</string>
    </property>
    <addaction name="action_open_db_manager"/>
    <addaction name="action_save_to_db_2"/>
   </widget>
   <addaction name="menu"/>
   <addaction name="menu_2"/>
  </widget>
  <action name="action_new">
   <property name="text">
    <string>Новая система</string>
   </property>
  </action>
  <action name="action_open">
   <property name="text">
    <string>Загрузить из файла...</string>
   </property>
  </action>
  <action name="action_save">
   <property name="text">
    <string>Сохранить</string>
   </property>
  </action>
  <action name="action_saveas">
   <property name="text">
    <string>Сохранить как...</string>
   </property>
  </action>
  <action name="action_exit">
   <property name="text">
    <string>Выход</string>
   </property>
  </action>
  <action name="action_open_db_manager">
   <property name="text">
    <string>Операция с БД...</string>
   </property>
  </action>
  <action name="action_save_to_db">
   <property name="text">
    <string>Сохранить в базу</string>
   </property>
  </action>
  <action name="action_save_to_db_2">
   <property name="text">
    <string>Сохранить в БД</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
 <buttongroups>
  <buttongroup name="buttonGroup_rotation_mode"/>
 </buttongroups>
</ui>
